src/fides/api/schemas/connection_configuration/connection_secrets_snowflake.py

```diff
@@ -37,20 +58,33 @@ class SnowflakeSchema(ConnectionConfigSecretsSchema):
         description="The name of the Snowflake schema within the selected database.",
     )
     role_name: Optional[str] = Field(
-        None,
         title="Role",
+        default=None,
         description="The Snowflake role to assume for the session, if different than Username.",
     )
 
     _required_components: List[str] = [
         "account_identifier",
         "user_login_name",
-        "password",
         "warehouse_name",
         "database_name",
         "schema_name",
     ]
 
+    @root_validator()
+    def validate_private_key_and_password(cls, values: dict) -> dict:
+        private_key: str = values.get("private_key", "")
+
+        if values.get("password") and values.get("private_key"):
```

Comment:
nit to use the local var 
```suggestion
        if values.get("password") and private_key:
```